src/usecase_regression.Rmd

```diff
@@ -0,0 +1,87 @@
+# Regression
+For the regression use case we use the well-known [Boston Housing](&mlbench::BostonHousing) dataset. The road map is as follows:
+
+* define the learning task  ([here](task.md)),
+* tune the model ([here](tune.md)),
+* conduct a benchmark experiment ([here](benchmark_experiments.md)) and
+* evaluate the performance of the model ([here](performance.md)).
+
+First, let's have a look at the data.
+```{r}
+data(BostonHousing, package = "mlbench")
+summary(BostonHousing)
+```
+This data set concerns housing in the suburban area of Boston. The target variable, chosen for the regression task, is ``medv`` - the median value of owner-occupied homes in $\$1000$'s. Description of the others $13$ attributes can be found [here](&mlbench::BostonHousing).
+
+### Define a task
+Now, let us continue with defining the regression [task](task.md).
+
+```{r}
+## Make a task
+regr.task = makeRegrTask(data = BostonHousing, target = "medv")
+regr.task
+```
+In order to get an overview of the features type, we can print out the ``regr.task``. This shows that there are $12$ numeric and one factor variables in the data set.
+
+### Tuning
+By calling ``listLearners("regr")`` we can see which learners are available for the regression task. We picked the classical linear regression model (``regr.lm``), the [Lasso](&penalized::penalized) regression model (``regr.penalized.lasso``), [SVM](&kernlab::ksvm) with a radial basis kernel (``regr.ksvm``) and random forest from the [ranger](&ranger::ranger) package (``regr.ranger``), which we would like to compare. Before setting up a [benchmark experiment](benchmark_experiments.md) we can specify which hyperparameters are going to be tuned. The [%mlr] package provides powerful tuning algorithms. See [Tuning](tuning.md) and [Advanced Tuning](advanced_tune.md) for more details.
```

Comment:
> , which we would like to compare

I would put this at the beginning of the sentence. Maybe something like

With so many learners it is difficult to choose which one would be optimal for this specific task. As such we will choose a sample of these learners and compare their results. This analysis uses...